packages/auth/src/AuthProvider.tsx

```diff
@@ -1,24 +1,276 @@
-﻿import { AuthContextValuesParams } from './types';
-import React from 'react';
-import { AuthContext, authContextValues } from './useAuth';
+﻿import React, {
+  useState,
+  useMemo,
+  useEffect,
+  useContext,
+  createContext,
+  useCallback,
+} from "react";
+import { Auth } from "@aws-amplify/auth";
+import { DataStore } from "@aws-amplify/datastore";
+import { Hub } from "@aws-amplify/core";
+import type { HubCallback } from "@aws-amplify/core";
+import { CognitoUser, CodeDeliveryDetails } from "amazon-cognito-identity-js";
+import {
+  ANSWER_CHALLENGE_ERRORS,
+  SIGN_IN_ERROR_CODES,
+  SIGN_IN_OR_CREATE_ACTIONS,
+  AuthContextValues,
+  AuthContextValuesParams,
+  CognitoUserWithAttributes,
+  SignInOrCreateResponse,
+} from "./types";
+import { MD5 } from "crypto-js";
 
 interface AuthProviderProps extends AuthContextValuesParams {
   children: JSX.Element | JSX.Element[];
 }
 
+const AuthContext = createContext<AuthContextValues>({
+  cognitoUser: undefined,
+  authenticated: false,
+  signInOrCreateUser: () => Promise.resolve({ action: "SignUp" }),
+  resendSignUp: () => Promise.resolve(undefined),
+  confirmSignUp: () => Promise.resolve({ success: false }),
+  confirmSignIn: () => Promise.resolve({ success: false }),
+  signOutUser: () => Promise.resolve(),
+  updateUserAttributes: () => Promise.resolve(),
+  userAttributes: null,
+});
+
 export const AuthProvider: React.FC<AuthProviderProps> = ({
   onSessionStart,
   onSessionFailed,
   children,
 }) => {
-  const providerValues = authContextValues({
-    onSessionStart,
-    onSessionFailed,
-  });
-
-  return (
-    <AuthContext.Provider value={providerValues}>
-      {children}
-    </AuthContext.Provider>
+  const [authenticated, setAuthenticated] = useState(false);
+  const [cognitoUser, setCognitoUser] = useState<CognitoUserWithAttributes>();
+  const [cognitoUserSignIn, setCognitoUserSignIn] = useState<CognitoUser>();
+
+  const handleSessionStart = useCallback(async () => {
```

Comment:
don't need the async anymore

Reply:
removed async on handleSessionFailed but this one will still need it as we need to update cognitouser on handleSessionStart.